src/pages/index.js

```diff
@@ -3,16 +3,36 @@ import CommunityLogo from '../assets/Community-Learning.svg';
 import EventCard from '../components/event-card';
 import Layout from '../components/layout';
 import SEO from '../components/seo';
-import mockEvents from '../mock-data/mockEvents';
 
 function IndexPage() {
   const [upcomingEvents, setUpcomingEvents] = useState([]);
 
   useEffect(() => {
-    const getEvents = async () => {
-      setUpcomingEvents(mockEvents);
-    };
-    getEvents();
+    console.log(window.location.href);
+    fetch(`${window.location.href}.netlify/functions/pingEvents`)
+      .then(res => res.json())
+      .then(data => {
+        let date_regex = /\w+(day), \w+ \d{1,2} (at) \d{1,2}:\d{1,2} \w+/g; // assumes this is the standard date format and will not change
+        let id_regex = /\w+(?=\/$)/g; // gets the id at the end of the url
+        let events = [];
+        let { items } = data;
+        for (let i = 0; i < 3; i++) {
+          // only need to display the three most upcoming events
+          let date_matches = items[i].content.match(date_regex);
+          let id_matches = items[i].guid.match(id_regex);
+
+          let event = {
+            id: id_matches ? id_matches[0] : 'no id found...',
+            time: date_matches ? date_matches[0] : 'no date found...',
```

Comment:
We're going to need better fallback code if the regex doesn't match. Time doesn't matter so much, but if the ID isn't working we'll want to not display the events or display an error.